### ISSUE TITLE
What is for me StarSpace Or Fasttext

### ISSUE THREAD

**Issue by beshoo** [facebookresearch/StarSpace#125]
First of all thank you so much for this wonderful lib.

Now i am using FastText for classification . i have 300k first name labeled ,Male and Female.
It return good results.

**How to stay In memory ?**
https://github.com/loretoparisi/fasttext.js/
Inorder to keep fastetx in memory and call it via http://localhost so i dont have to load the lib into memory eachtime and it is easy to call localhost via PHP scripts hosted on the same server.

**Is StarSpace better than FastText**
When i see neural model  word it come to my head that this is some kind Of AI , and it will be more powerful than Ngram and other NLP trikes.

I am php programmer , and i am not that much in AI and R and Prediction algorithm. so what is your advice.

1 - How neural model learn form new unlabeled data , that the question.... 
What if i have 300k labeled  first name data-set and i have another 100m unlabeled datastet , well StarSpace learn and improve it self ?

2- Do i have to use the  StarSpace over fastText  for this text NLP calcification ?

3- How can i use StarSpace  from memory to call it via CURL as example! 
eg. https://github.com/loretoparisi/fasttext.js/

Regards

**Comment by beshoo** [comment]
Update : 
I used this settings (I am testing) with StarSpace

Arguments:
```
lr: 0.01
dim: 10
epoch: 100
maxTrainTime: 8640000
saveEveryEpoch: 0
loss: hinge
margin: 0.05
similarity: cosine
maxNegSamples: 10
negSearchLimit: 50
thread: 10
minCount: 1
minCountLabel: 1
label: __label__
ngrams: 1
bucket: 2000000
adagrad: 1
trainMode: 0
fileFormat: fastText
normalizeText: 0
dropoutLHS: 0
dropoutRHS: 0
Predictions use 2 known labels.
```
My dataset something like 

```
__label__female aabeer
__label__female aabha
__label__female aabha
__label__female aabia
__label__female aabid
__label__female aabida
__label__female aabidah
__label__male Jorge
__label__male Орал
__label__male Smae
__label__male Павло
```

What is the best settings for this kind of datastet ? 
Problem  1: I want to pass a first name and predict Gender.
Problem  2 : How can i make this data relearn over time form the new unlabeled data?

**Comment by ledw** [comment]
@beshoo Thanks for your questions. On your particular problem, I would expect fastText and StarSpace perform similarly. StarSpace has the advantage of handling featured labels comparing to fastText, but it is not your case here. 
To find the best parameters for your task, you can try different similarity (dot, cosine), loss function (hinge, softmax), negSearchLimit, maxNegSamples, learning rate, etc.
Currently we do not have support for calling from php yet.

**Comment by ledw** [comment]
Closing old issue.

**Comment by pganeshGapps** [comment]
@ledw  Can you please elaborate more on : "StarSpace has the advantage of handling featured labels comparing to fastText"?
What do you mean by 'featured labels' ?

**Comment by ledw** [comment]
@pganeshGapps Hi, featured labels means that one can use features to represent labels. For instance, in StarSpace label can be a sentence, which is represented by the words in the sentence (bag of words, etc.). While in fastText label needs to be direct embeddings like words or tags.
